src/main/groovy/com/codebrig/arthur/observe/structure/filter/operator/relational/compare/IsEqualOperatorFilter.groovy

```diff
@@ -27,6 +28,10 @@ class IsEqualOperatorFilter extends StructureFilter<IsEqualOperatorFilter, Void>
 
     @Override
     boolean evaluate(SourceNode node) {
-        return filter.evaluate(node)
+        if (Objects.requireNonNull(node).internalType == "CPPASTBinaryExpression" && node.token == "==") {
```

Comment:
Same thing. Filters must check that it's not-null if that's part of its logic, but they can't require it not to be null.